hack/deploy-kubesphere.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The KubeSphere Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+function wait_for_ks-installer_pod() {
+    counter=0
+    while [  $counter -lt 20 ] ;do
+        log=$(kubectl -n kubesphere-system get pod | grep ks-installer)
+        if [[ $log =~ "Running" ]]
+            then
+                break
+        fi
+        echo "waiting ks-installer pod for ready"
+    sleep 10
+    done
+}
+
+function wait_for_ks-installer_finish() {
+    while IFS= read -r line; do
+        if [[ $line =~ "Welcome to KubeSphere" ]]
+            then
+                break
+        fi
+    done < <(timeout 900 kubectl logs -n kubesphere-system deploy/ks-installer -f)
+}
+
+IMAGE_NAME="${1:-}"
+
+# use kubespheredev and latest tag as default image
+TAG="${TAG:-latest}"
+REPO="${REPO:-kubespheredev}"
+
+# Use KIND_LOAD_IMAGE=y .hack/deploy-kubesphere.sh to load
+# the built docker image into kind before deploying.
+if [[ "${KIND_LOAD_IMAGE:-}" == "y" ]]; then
+    kind load docker-image "$REPO/ks-apiserver:$TAG" --name="${KIND_CLUSTER_NAME:-kind}"
+    kind load docker-image "$REPO/ks-controller-manager:$TAG" --name="${KIND_CLUSTER_NAME:-kind}"
+fi
+
+# download the latest ks-install to deploy KubeSphere
+wget https://raw.githubusercontent.com/kubesphere/ks-installer/master/deploy/kubesphere-installer.yaml
+wget https://raw.githubusercontent.com/kubesphere/ks-installer/master/deploy/cluster-configuration.yaml
+
+kubectl apply -f kubesphere-installer.yaml
+kubectl apply -f cluster-configuration.yaml
```

Comment:
why not 
```
kubectl apply -f https://raw.githubusercontent.com/kubesphere/ks-installer/master/deploy/kubesphere-installer.yaml
kubectl apply -f https://raw.githubusercontent.com/kubesphere/ks-installer/master/deploy/cluster-configuration.yaml
```

Reply:
Also, I think we need to check if `kubectl apply` successfully in case of network issues